src/features/counterfactual/hooks/usePendingSafeNotifications.ts

```diff
@@ -26,6 +29,8 @@ const usePendingSafeNotifications = (): void => {
   const dispatch = useAppDispatch()
   const chain = useCurrentChain()
   const safeAddress = useSafeAddress()
+  const { address = '' } = useWallet() || {}
+  const { refetch } = useGetAllOwnedSafesQuery(address === '' ? skipToken : { walletAddress: address })
```

Comment:
Its probably cleaner to create a new listener middleware for this. Wdyt @katspaugh @compojoom

Reply:
Although it does require access to the connected wallet which I am not sure is possible inside a middleware. We would have to move the connected wallet state into Redux as well.